packages/relic/lib/src/headers/special/accept_encoding_header.dart

```diff
@@ -0,0 +1,55 @@
+part of '../headers.dart';
+
+/// A class representing the HTTP Accept-Encoding header.
+///
+/// This header specifies the content encoding that the client can understand.
+class AcceptEncodingHeader {
+  /// The list of encodings that are accepted.
+  final List<EncodingQuality> encodings;
+
+  /// Constructs an instance of [AcceptEncodingHeader] with the given encodings.
+  AcceptEncodingHeader(this.encodings);
+
+  /// Parses the Accept-Encoding header value and returns an [AcceptEncodingHeader] instance.
+  factory AcceptEncodingHeader.parse(List<String> values) {
+    var parts = values.splitTrimAndFilterUnique();
+
+    if (parts.isEmpty) {
+      throw FormatException(
+        'Invalid Accept-Encoding header: value cannot be empty',
+      );
+    }
+
+    var encodings = parts.map((e) {
+      var parts = e.split(';q=');
+      var encoding = parts[0].trim().toLowerCase();
+      var quality = switch (parts.length > 1) {
+        true => double.parse(parts[1].trim()),
+        false => 1.0,
+      };
+      return EncodingQuality(encoding, quality);
+    }).toList();
+
+    return AcceptEncodingHeader(encodings);
+  }
+
+  /// Converts the [AcceptEncodingHeader] instance into a string representation suitable for HTTP headers.
+  @override
+  String toString() => encodings.map((e) => e.toString()).join(', ');
+}
+
+/// A class representing an encoding with an optional quality value.
+class EncodingQuality {
+  /// The encoding value.
+  final String encoding;
+
+  /// The quality value (default is 1.0).
+  final double quality;
+
+  /// Constructs an instance of [EncodingQuality].
+  EncodingQuality(this.encoding, [this.quality = 1.0]);
```

Comment:
Suggestion: Make quality a nullable parameter and don't assign it on the outside.

Since we have logic (in the toString) statement regarding the value of the quality, we should keep that codification on the inside.

Reply:
like this:  `EncodingQuality(this.encoding, [double? quality]) : quality = quality ?? 1.0;` ?

Reply:
Yes, that makes sense! 👍